pkg/k8s/apis/cilium.io/v2/validator/validator.go

```diff
@@ -18,21 +18,27 @@ import (
 	"encoding/json"
 	"fmt"
 
+	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
 	"github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2/client"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/policy/api"
 	"github.com/go-openapi/validate"
+	"github.com/sirupsen/logrus"
 	apiextensionsinternal "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
 	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
 	"k8s.io/apiextensions-apiserver/pkg/apiserver/validation"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
-// NPValidator is a validator structure used to validate CNP.
+// NPValidator is a validator structure used to validate CNP and CCNP.
 type NPValidator struct {
 	cnpValidator  *validate.SchemaValidator
 	ccnpValidator *validate.SchemaValidator
+
+	logger *logrus.Logger
```

Comment:
Why is this necessary? Can we make it a package-level variable like how we do in other packages throughout the codebase? 

Removing this dependency-injection will also allow us to avoid instantiating the logger in the test code as well.

Reply:
Yeah, this makes sense actually. We can also attach `validator` subsystem context to the logs. I will update the PR.